go-controller/pkg/util/node_annotations.go

```diff
@@ -465,38 +487,51 @@ func ParseNodeGatewayRouterLRPAddr(node *kapi.Node) (net.IP, error) {
 	return ip, nil
 }
 
-// ParseNodeGatewayRouterLRPAddrs returns the IPv4 and/or IPv6 addresses for the node's gateway router port
-// stored in the 'ovnNodeGRLRPAddr' annotation
-func ParseNodeGatewayRouterLRPAddrs(node *kapi.Node) ([]*net.IPNet, error) {
-	nodeIfAddrAnnotation, ok := node.Annotations[ovnNodeGRLRPAddr]
+// parsePrimaryIfAddrAnnotation unmarshals the IPv4 / IPv6 values in the
+// primaryIfAddrAnnotation format from the nodeAnnotation map with the
+// provided 'annotationName' as key and returns the addresses.
+func parsePrimaryIfAddrAnnotation(node *kapi.Node, annotationName string) ([]*net.IPNet, error) {
```

Comment:
I think I named it because it is parsing an annotation which is of the format defined in the struct
type primaryIfAddrAnnotation struct {
	IPv4 string `json:"ipv4,omitempty"`
	IPv6 string `json:"ipv6,omitempty"`
}

here [https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/util/node_annotations.go#L328](url)